src/pages/Index/Aside/Recents/index.js

```diff
@@ -32,12 +50,26 @@ const Recents = () => {
     return null
   }
 
+  const Component = TAB_COMPONENT[tab]
+
   return (
     <Section title='Recently viewed'>
-      {!!assets.length && <Assets slugs={assets} />}
-      {!!chartLayouts.length && <ChartLayouts ids={chartLayouts} />}
-      {!!watchlists.length && <Watchlists ids={watchlists} />}
-      {!!screeners.length && <Watchlists title='Screeners' ids={screeners} />}
+      <Tabs
+        className={styles.tabs}
+        options={TABS}
+        defaultSelectedIndex={tab}
+        onSelect={tab => setTab(tab)}
+        classes={styles}
+      />
+
+      {Component && (
+        <Component
+          assets={assets}
+          chartLayouts={chartLayouts}
+          watchlists={watchlists}
+          screeners={screeners}
+        />
+      )}
```

Comment:
I am not sure but looks like even if a component doesn't contain items (for ex. no recent screeners), the tab will appear and will be just empty. Did you discuss with Anastasya that case and what we should show?

Reply:
Also here every component will receive items from other components, which won't use.

Reply:
Ok, will check such case